### PR TITLE
Add more tests to check c calls

### PR DESCRIPTION
**Pull request by eivindjahren**
Also fixes a bug in `Grid.compare`.

In order to be able to remove the c code safely, we have to ensure all `ReadataPrototype`s can still be called. This PR creates tests that tries to call as many of those as possible.

The list of covered resdataprotypes are as follows.

- [x] `python/resdata/__init__.py` (no ResdataPrototypes)
- [x] `python/resdata/bin.py` (no ResdataPrototypes)
- [x] `python/resdata/geometry/__init__.py` (no ResdataPrototypes)
- [x] `python/resdata/geometry/cpolyline.py`
- [x] `python/resdata/geometry/cpolyline_collection.py`
- [x] `python/resdata/geometry/geo_pointset.py`
- [x] `python/resdata/geometry/geo_region.py`
- [x] `python/resdata/geometry/geometry_tools.py` (no ResdataPrototypes)
- [x] `python/resdata/geometry/polyline.py` (no ResdataPrototypes)
- [x] `python/resdata/geometry/surface.py`
- [x] `python/resdata/geometry/xyz_io.py` (no ResdataPrototypes)
- [x] `python/resdata/gravimetry/__init__.py` (no ResdataPrototypes)
- [x] `python/resdata/gravimetry/rd_grav.py`
- [x] `python/resdata/gravimetry/rd_subsidence.py`
- [x] `python/resdata/grid/__init__.py` (no ResdataPrototypes)
- [x] `python/resdata/grid/cell.py` (no ResdataPrototypes)
- [x] `python/resdata/grid/faults/__init__.py` (no ResdataPrototypes)
- [x] `python/resdata/grid/faults/fault.py` (no ResdataPrototypes)
- [x] `python/resdata/grid/faults/fault_block.py`
- [x] `python/resdata/grid/faults/fault_block_layer.py`
- [x] `python/resdata/grid/faults/fault_collection.py` (no ResdataPrototypes)
- [x] `python/resdata/grid/faults/fault_line.py` (no ResdataPrototypes)
- [x] `python/resdata/grid/faults/fault_segments.py` (no ResdataPrototypes)
- [x] `python/resdata/grid/faults/layer.py`
- [x] `python/resdata/grid/rd_grid.py` (relying on equinor tests for lgr)
- [x] `python/resdata/grid/rd_grid_generator.py`
- [x] `python/resdata/grid/rd_region.py(no ResdataPrototypes)`
- [x] `python/resdata/rd_type.py`
- [x] `python/resdata/rd_util.py`
- [x] `python/resdata/resfile/__init__.py` (no ResdataPrototypes)
- [x] `python/resdata/resfile/fortio.py`
- [x] `python/resdata/resfile/rd_3d_file.py` (no ResdataPrototypes)
- [x] `python/resdata/resfile/rd_3dkw.py` (no ResdataPrototypes)
- [x] `python/resdata/resfile/rd_file.py`
- [x] `python/resdata/resfile/rd_file_view.py`
- [x] `python/resdata/resfile/rd_init_file.py` (no ResdataPrototypes)
- [x] `python/resdata/resfile/rd_kw.py`
- [x] `python/resdata/resfile/rd_restart_file.py`
- [x] `python/resdata/rft/__init__.py` (no ResdataPrototypes)
- [x] `python/resdata/rft/rd_rft.py` (partially tested through equinor data and difficult to produce data)
- [x] `python/resdata/rft/rd_rft_cell.py` (partially tested through equinor data and difficult to produce data)
- [x] `python/resdata/rft/well_trajectory.py` (no ResdataPrototypes)
- [x] `python/resdata/summary/__init__.py` (no ResdataPrototypes)
- [x] `python/resdata/summary/rd_cmp.py` (no ResdataPrototypes)
- [x] `python/resdata/summary/rd_npv.py` (no ResdataPrototypes)
- [x] `python/resdata/summary/rd_smspec_node.py`
- [x] `python/resdata/summary/rd_sum.py`
- [x] `python/resdata/summary/rd_sum_keyword_vector.py`
- [x] `python/resdata/summary/rd_sum_node.py` (no ResdataPrototypes)
- [x] `python/resdata/summary/rd_sum_tstep.py`
- [x] `python/resdata/summary/rd_sum_var_type.py` (no ResdataPrototypes)
- [x] `python/resdata/summary/rd_sum_vector.py` (no ResdataPrototypes)
- [x] `python/resdata/util/__init__.py` (no ResdataPrototypes)
- [x] `python/resdata/util/enums/__init__.py` (no ResdataPrototypes)
- [x] `python/resdata/util/enums/rng_alg_type_enum.py` (no ResdataPrototypes)
- [x] `python/resdata/util/enums/rng_init_mode_enum.py` (no ResdataPrototypes)
- [x] `python/resdata/util/test/__init__.py` (no ResdataPrototypes)
- [x] `python/resdata/util/test/debug_msg.py` (no ResdataPrototypes)
- [x] `python/resdata/util/test/extended_testcase.py` (no ResdataPrototypes)
- [x] `python/resdata/util/test/import_test_case.py` (no ResdataPrototypes)
- [x] `python/resdata/util/test/lint_test_case.py` (no ResdataPrototypes)
- [x] `python/resdata/util/test/mock/__init__.py` (no ResdataPrototypes)
- [x] `python/resdata/util/test/mock/rd_sum_mock.py` (no ResdataPrototypes) 
- [x] `python/resdata/util/test/path_context.py` (no ResdataPrototypes)
- [x] `python/resdata/util/test/resdata_test_runner.py`(no ResdataPrototypes) 
- [x] `python/resdata/util/test/source_enumerator.py` (no ResdataPrototypes)
- [x] `python/resdata/util/test/test_area.py`
- [x] `python/resdata/util/test/test_run.py` (no ResdataPrototypes)
- [x] `python/resdata/util/util/__init__.py` (no ResdataPrototypes)
- [ ] `python/resdata/util/util/bool_vector.py`
- [ ] `python/resdata/util/util/cthread_pool.py` (current plan is to remove it)
- [x] `python/resdata/util/util/ctime.py`
- [x] `python/resdata/util/util/cwd_context.py` (no ResdataPrototypes)
- [ ] `python/resdata/util/util/double_vector.py` (through vector_template.py)
- [x] `python/resdata/util/util/hash.py`
- [ ] `python/resdata/util/util/install_abort_signals.py`
- [ ] `python/resdata/util/util/int_vector.py` (through vector_template.py)
- [x] `python/resdata/util/util/lookup_table.py`
- [x] `python/resdata/util/util/permutation_vector.py`
- [x] `python/resdata/util/util/rng.py`
- [x] `python/resdata/util/util/stringlist.py`
- [x] `python/resdata/util/util/thread_pool.py` (no ResdataPrototypes)
- [ ] `python/resdata/util/util/time_vector.py`  (through vector_template.py)
- [x] `python/resdata/util/util/vector_template.py`
- [x] `python/resdata/util/util/version.py`
- [x] `python/resdata/version.py` (no ResdataPrototypes)
- [x] `python/resdata/well/__init__.py` (no ResdataPrototypes)
- [x] `python/resdata/well/well_connection.py`
- [x] `python/resdata/well/well_connection_direction_enum.py` (no ResdataPrototypes)
- [x] `python/resdata/well/well_info.py`
- [ ] `python/resdata/well/well_segment.py` (only through equinor data)
- [x] `python/resdata/well/well_state.py` (self._segment_collection_iget is called with equinor data)
- [x] `python/resdata/well/well_time_line.py`
- [x] `python/resdata/well/well_type_enum.py` (no ResdataPrototypes)

Of special mention is the vector classes which are tested quite a bit, but the c code is also generated from a template, so although all are not strictly covered, it is not that important.

The few remaining untested `ResdataPrototype`s are a bit tricky to get to, and it is perhaps not worth it to spend time on it when there is some coverage of it through the equinor testdata.

### CODE REVIEW

File: python/tests/rd_tests/test_grav.py [hunk 1]
```diff
@@ -1,24 +1,100 @@
...
+                header[66] = 2010
+
+                seq_hdr.fwrite(f)
+                header.fwrite(f)
+                for kw in kws:
+                    kw.fwrite(f)
+
+            # The init file created here only contains a PORO field. More
+            # properties must be added to this before it can be used for
+            # any usefull gravity calculations.
```
Comment (berland) [inline python/tests/rd_tests/test_grav.py:77]: usefull -> useful

File: python/tests/rd_tests/test_rd_util.py [hunk 2]
```diff
@@ -18,3 +21,32 @@ def test_file_report_nr(self):
...
+            f"""\
+            PARALLEL
+            {data_file_num_cpu} DISTRIBUTED/
+            """
+        )
+    )
+    assert ResdataUtil.get_num_cpu(str(data_file)) == 4
+
+
+def test_num_cpu_from_data_file_used_if_config_num_cpu_not_set(tmp_path):
```
Comment (berland) [inline python/tests/rd_tests/test_rd_util.py:40]: This test names seems incorrect

Comment (eivindjahren) [reply python/tests/rd_tests/test_rd_util.py]: Fixed

File: python/tests/rd_tests/test_grav.py (hunk 1 above)
Comment (eivindjahren) [reply python/tests/rd_tests/test_grav.py]: Fixed

File: test-data/local/ECLIPSE/simple/SIMPLE.DATA [hunk 3]
```diff
@@ -0,0 +1,59 @@
+RUNSPEC
+START
+1 JAN 2017 /
+
+DIMENS
+ 10 10 10 /
+OIL
+UNIFOUT
```
Comment (berland) [inline test-data/local/ECLIPSE/simple/SIMPLE.DATA:8]: No syntax erorr, but I would not separate the line with OIL and the line with WATER with UNIFOUT. Humans need to view line 7 and 9 together to figure out which phase configuration is active.

File: test-data/local/ECLIPSE/simple/SIMPLE.DATA [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+PVTW
+ 400    1    1.0E-06    1      0 /
+PVTO
+  0.001 15.0  1.0 0.1 /
+  1.6 5000.0  1.8 0.5
+      9000.0  1.7 0.6 /
+/
+SOLUTION
+EQUIL
+ 3*1000 0 1000 0 0 0 0 /
```
Comment (berland) [inline test-data/local/ECLIPSE/simple/SIMPLE.DATA:51]: High pressure at shallow depths :) 

If it does not matter for values in the test you can have 
`1000 100 1000 /` 
instead here to have something slightly more meaningful (item 5 is not relevant in this oil-water problem)

Comment (eivindjahren) [reply test-data/local/ECLIPSE/simple/SIMPLE.DATA:51]: So in total:
```
3*1000 0 1000 1000 100 1000 /
```
?

Comment (eivindjahren) [reply test-data/local/ECLIPSE/simple/SIMPLE.DATA]: I think I changed it to what you wanted now.

File: test-data/local/ECLIPSE/simple/SIMPLE.DATA (hunk 3 above)
Comment (eivindjahren) [reply test-data/local/ECLIPSE/simple/SIMPLE.DATA:7]: Fixed

**Review by berland — APPROVED** [review]
Nice!

**Pull request merged**